src/cli/commands/append/expireTilesFilters.ts

```diff
@@ -0,0 +1,38 @@
+import { BoundingBox } from '@map-colonies/tile-calc';
+import { booleanContains, Feature, Geometry, Polygon } from '@turf/turf';
+
+export type ExpireTilesPreFilterFunc = (expireList: string[]) => string[];
+export type ExpireTilesPostFilterFunc = (bbox: BoundingBox[]) => BoundingBox[];
+
+export const filterByZoom = (zoom: number): ExpireTilesPreFilterFunc => {
+  const filter: ExpireTilesPreFilterFunc = (expireList) => {
+    return expireList.filter((expireTileLine) => {
+      const elements = expireTileLine.split('/');
+      const zoomValue = parseInt(elements[0]);
+      return zoomValue === zoom;
+    });
+  };
+  return filter;
+};
+
+export const filterByGeometry = (geometry: Feature | Geometry): ExpireTilesPostFilterFunc => {
+  const filter: ExpireTilesPostFilterFunc = (bbox) => {
+    return bbox.filter((bboxItem) => {
+      const { west, south, east, north } = bboxItem;
+      const bboxAsPolygon: Polygon = {
+        type: 'Polygon',
+        coordinates: [
+          [
+            [west, south],
+            [east, south],
+            [east, north],
+            [west, north],
+            [west, south],
+          ],
+        ],
+      };
+      return booleanContains(geometry, bboxAsPolygon);
+    });
+  };
+  return filter;
```

Comment:
if it efficient since you are transforming the bbox to polygon?

Reply:
I've improved the filtering chain functionality,
we get the w\s\e\n with `@mapbox/sphericalmercator` `bbox` function.
it is possible filter the tile before creating the whole bbox array but it losses the generic functionality of building the bbox and filtering them separately, for instance, say in the future we'd like to match some filters and not all.
also, it just seems less readable